src/treq/api.py

```diff
@@ -116,10 +119,19 @@ def request(method, url, **kwargs):
 
 def _client(*args, **kwargs):
     agent = kwargs.get('agent')
+    proxy = kwargs.get('proxy')
+    reactor = default_reactor(kwargs.get('reactor'))
+    pool = default_pool(reactor,
+                        kwargs.get('pool'),
+                        kwargs.get('persistent'))
+    if not agent and proxy is not None:
+        (address, port) = proxy
+        endpoint = HostnameEndpoint(
```

Comment:
probably not. I don't use proxies myself that often and I've mostly just been trying to get the PR to an acceptable point, it actually almost doesn't reflect the original code anymore. Perhaps we want a better overall solution here? I think authentication can come in various forms though - which would complicate this ...